### PR TITLE
feat(command): ✨  add Stdin arg

### LINKED ISSUE #43: Add support for stdin. Send file contents (string) to stdin of the process
## Hello!

- Vote on this issue by adding a 👍 reaction
- If you want to implement this feature, comment to let us know (we'll work with you on design, scheduling, etc.)

## Issue details

I need to be able to "send" a potentially large amount of data (a file) to an external process. It's cumbersome to try to encode the contents and send that as an "arg" to the command. Some commands may not even support that, they may only support a file on disk or `stdin` for input.

This is a request to accept a new parameter that takes a `string` (essentially the contents of the file) and passes it as `stdin` to the command.

### Affected area/feature

Local command resource

### PR DESCRIPTION
**Pull request by ghostsquad**
allows passing a string (think file contents) as stdin to a command

resolves #43

### CODE REVIEW

**Comment by lukehoban** [comment]
Thanks for the PR @ghostsquad!

A few questions and additional things to add in here:
1. Should `stdin` be supported for both local and remote commands?  Ideally, we should keep consistency in the APIs between the two where the concept makes sense for both, and I believe this can make sense for both local and remote Commands.
2. There will need to be updates to `schema.json` to include the new input property.  This is what will ensure this new property is generated in the various language SDKs.  After updating `schema.json`, you will need to run `make gen` to generate the SDK updates and include them in the PR.
3. We should add tests as part of this PR.  Something similar to the `examples/random` test case, but taking advantage of `stdin`.  The test itself can be added similar to https://github.com/pulumi/pulumi-command/blob/master/examples/examples_nodejs_test.go#L21:L32.

**Comment by ghostsquad** [comment]
@lukehoban I've made the changes you requested. As a new contributor to the code base, there were a few things I needed to install and configure in order to regenerate all the SDKs. I've added a separate commit with those changes. I know this is a bit different from the [CONTRIBUTE guide](https://github.com/pulumi/pulumi/blob/master/CONTRIBUTING.md), but I thought it might help someone else install/manage the required tools & binaries a bit easier.

If you do not wish to keep that commit, let me know, and I will remove.

Along this note, I was also considering updating `.github/workflows` with:

```yaml
- uses: asdf-vm/actions/setup@v1
- uses: asdf-vm/actions/install@v1
```

and some other changes in order to really unify the local development process with the github actions workflow process, but I felt that was a bit overreaching.

**Comment by lukehoban** [comment]
> As a new contributor to the code base, there were a few things I needed to install and configure in order to regenerate all the SDKs. I've added a separate commit with those changes. I know this is a bit different from the [CONTRIBUTE guide](https://github.com/pulumi/pulumi/blob/master/CONTRIBUTING.md), but I thought it might help someone else install/manage the required tools & binaries a bit easier.

I love the work here to try to make this more contribution friendly.  That said:
1. I'd suggest we move this into it's own PR for now, since it is an orthogonal change.
2. We have a general goal to keep the contribution guidelines, build targets, etc. reasonably consistent across all the Pulumi repos, and in particular across all the Pulumi package repos (providers, component packages, etc.).  I think we probably want to take changes like the ones here either ~everywhere, or else not take them, so that we can maintain consistency.  I think opening a PR with a proposed set of changes here, and having @stack72 and @guineveresaenger review for which pieces make sense to incorporate broadly would be a good way to figure out what we should take here.

**Comment by ghostsquad** [comment]
It looks like the failed tests are due to missing AWS credentials because this is a pull request.

**Comment by ghostsquad** [comment]
I missed changing this:

https://github.com/pulumi/pulumi-command/pull/44/commits/44c1b4d05f397b630a2f14dbfc1ec3116a349698#diff-418e6e7d93aeacebe7c22d5fb674b48bfa15c6a514eb3ae4744b86d447595d8e

It's a bit unfortunate that the list of fields is maintained in several different places.

But I think this is ready for re-review at this point.

File: examples/examples_go_test.go [hunk 1]
```diff
@@ -25,6 +25,19 @@ func TestRandomGo(t *testing.T) {
...
 }
 
+func TestStdinGo(t *testing.T) {
+	test := getGoBaseOptions(t).
+		With(integration.ProgramTestOptions{
+			Dir: filepath.Join(getCwd(t), "stdin-go"),
+			ExtraRuntimeValidation: func(t *testing.T, stack integration.RuntimeValidationStackInfo) {
+				out, ok := stack.Outputs["output"].(string)
+				assert.True(t, ok)
+				assert.Equal(t, "the quick brown fox")
```
[the remark this answers is not in the document]
Comment (ghostsquad) [reply examples/examples_go_test.go:35]: Oh that's embarrassing 😂

[the remark this answers is not in the document]
Comment (lukehoban) [reply examples/examples_go_test.go:35]: :-).  Unfortunately doesn't look like I can push to your fork, but I did validate that after this fix, I can run the tests successfully locally.  So should be good to merge once this is fixed up.

File: examples/examples_nodejs_test.go [hunk 2]
```diff
@@ -31,6 +31,19 @@ func TestRandom(t *testing.T) {
...
 }
 
+func TestStdin(t *testing.T) {
+	test := getJSBaseOptions(t).
+		With(integration.ProgramTestOptions{
+			Dir: filepath.Join(getCwd(t), "stdin"),
+			ExtraRuntimeValidation: func(t *testing.T, stack integration.RuntimeValidationStackInfo) {
+				out, ok := stack.Outputs["output"].(string)
+				assert.True(t, ok)
+				assert.Equal(t, "the quick brown fox")
```
[the remark this answers is not in the document]
Comment (ghostsquad) [reply examples/examples_nodejs_test.go]: fixed!

**Comment by ghostsquad** [comment]
thanks @lukehoban fixed those tests

**Comment by lukehoban** [comment]
Thanks @ghostsquad!

**Pull request merged**